File: cosmic-journal/cosmic-journal-backend/app/utils/locationUtils.py
import requests
from fastapi import HTTPException

def get_coordinates(city: str, api_key: str) -> tuple[float, float]:
    """Get latitude and longitude for a city using Geoapify."""
    url = f"https://api.geoapify.com/v1/geocode/search?text={city}&apiKey={api_key}"
    response = requests.get(url)
    if response.status_code != 200:
        raise HTTPException(status_code=400, detail="Failed to fetch coordinates")
    data = response.json()
    if not data["features"]:
        raise HTTPException(status_code=400, detail="City not found")
    return (
        data["features"][0]["properties"]["lat"],
        data["features"][0]["properties"]["lon"],
    )

File: cosmic-journal-frontend/src/utils/locationUtils.js
export async function getCoordinates(city, apiKey) {
  try {
    const response = await fetch(
      `https://api.geoapify.com/v1/geocode/search?text=${encodeURIComponent(city)}&apiKey=${apiKey}`
    );
    const data = await response.json();
    if (data.features.length === 0) throw new Error('City not found');
    return {
      latitude: data.features[0].properties.lat,
      longitude: data.features[0].properties.lon,
    };
  } catch (error) {
    throw new Error(`Failed to get coordinates: ${error.message}`);
  }
}